pkg/resource/configurator.go

```diff
@@ -41,33 +40,30 @@ func (cc ConfiguratorChain) Configure(ctx context.Context, cm Claim, cs Class, m
 }
 
 // An ObjectMetaConfigurator sets standard object metadata for a dynamically
-// provisioned resource, deriving it from a class and claim.
-type ObjectMetaConfigurator struct {
-	typer runtime.ObjectTyper
-}
+// provisioned resource, deriving it from a class and claim. It is deprecated;
+// use ConfigureNames instead.
+type ObjectMetaConfigurator struct{}
 
 // NewObjectMetaConfigurator returns a new ObjectMetaConfigurator.
-func NewObjectMetaConfigurator(t runtime.ObjectTyper) *ObjectMetaConfigurator {
-	return &ObjectMetaConfigurator{typer: t}
+func NewObjectMetaConfigurator(_ runtime.ObjectTyper) *ObjectMetaConfigurator {
```

Comment:
I like how this makes it such that `NewObjectMetaConfigurator` code does not have to be modified! What do you think our policy for removing deprecated code here should be given this library is pre-v1.0?

Reply:
As far as I can tell https://semver.org/ does not restrict us in any way while we're below version 1. I think a good compromise would be:

1. Don't break API compatibility on a patch release.
1. It's okay to mark an API as deprecated on a patch release.
1. A little subjective, but it's okay to make trivial behaviour changes on a patch release.

Reply:
IMHO, it should be fine to release a deprecation notice in a release and after that one break the function signature directly. The consumers will see the error on build time, so, it should be fine and fairly easy for them to fix it.